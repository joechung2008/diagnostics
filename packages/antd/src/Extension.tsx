import { Typography } from "antd";
import Configuration from "./Configuration";
import StageDefinition from "./StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div className="extension-root grow">
      <Typography.Title level={1}>{extensionName}</Typography.Title>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
