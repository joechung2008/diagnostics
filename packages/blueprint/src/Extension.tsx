import Configuration from "./Configuration";
import StageDefinition from "./StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div id="extension" className="extension-root grow">
      <h2 className="bp6-heading">{extensionName}</h2>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
