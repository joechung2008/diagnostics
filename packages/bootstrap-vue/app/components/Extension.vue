<template>
  <div class="extension">
    <h1>{{ extensionName }}</h1>
    <Configuration v-if="config" :config="config" />
    <StageDefinition v-if="stageDefinition" :stage-definition="stageDefinition" />
  </div>
</template>

<script setup lang="ts">
import type { ExtensionProps } from "@/types/Extension";

defineProps<ExtensionProps>();
</script>

<style scoped>
.extension {
  display: flex;
  flex: 1;
  flex-direction: column;
  gap: 0.5rem;
  max-height: max-content;
  overflow-y: auto;
  padding: 0.25rem;
}
</style>
