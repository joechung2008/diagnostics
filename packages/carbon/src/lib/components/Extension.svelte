<script lang="ts">
	import Configuration from './Configuration.svelte';
	import StageDefinition from './StageDefinition.svelte';

	let { extensionName, config, stageDefinition }: ExtensionProps = $props();
</script>

<div class="extension-root" role="tabpanel" tabindex="0">
	<h1>{extensionName}</h1>
	{#if config}
		<Configuration {config} />
	{/if}
	{#if stageDefinition}
		<StageDefinition {stageDefinition} />
	{/if}
</div>

<style>
	.extension-root {
		display: flex;
		flex: 1;
		flex-direction: column;
		gap: 0.5rem;
		overflow-y: auto;
		padding: 0.5rem;
	}
</style>
