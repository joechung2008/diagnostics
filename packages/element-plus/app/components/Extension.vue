<template>
    <div class="extension">
        <h1>{{ extensionName }}</h1>
        <Configuration v-if="config" :config="config" />
        <StageDefinition v-if="stageDefinition" :stage-definition="stageDefinition" />
    </div>
</template>

<script setup lang="ts">
import type { ExtensionProps } from "@/types"

defineProps<ExtensionProps>()
</script>

<style scoped>
.extension {
    display: flex;
    flex: 1;
    flex-direction: column;
    gap: 0.5rem;
    max-height: max-content;
    overflow-y: auto;
    padding: 0.25rem;
}

h1 {
    margin: 0;
}
</style>
