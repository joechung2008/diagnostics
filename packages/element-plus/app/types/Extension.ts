import type { ExtensionInfo } from "./ExtensionInfo"
import type { ExtensionError } from "./ExtensionError"

export type Extension = ExtensionInfo | ExtensionError
