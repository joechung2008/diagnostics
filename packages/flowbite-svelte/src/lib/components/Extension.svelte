<script lang="ts">
  import { Heading } from "flowbite-svelte";
  import Configuration from "./Configuration.svelte";
  import StageDefinition from "./StageDefinition.svelte";

  let { config, extensionName, stageDefinition }: ExtensionProps = $props();
</script>

<div class="flex flex-grow flex-col gap-0.5 overflow-y-auto p-1">
  <Heading tag="h1">
    {extensionName}
  </Heading>
  {#if config}
    <Configuration {config} />
  {/if}
  {#if stageDefinition}
    <StageDefinition {stageDefinition} />
  {/if}
</div>
