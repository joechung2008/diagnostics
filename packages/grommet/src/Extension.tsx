import { Heading } from "grommet";
import Configuration from "./Configuration";
import StageDefinition from "./StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div className="extension-root grow">
      <Heading level={2} margin="none">
        {extensionName}
      </Heading>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
