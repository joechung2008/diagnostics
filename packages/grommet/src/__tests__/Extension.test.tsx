import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import Extension from "../Extension";

describe("Extension", () => {
  it("renders with only extensionName", () => {
    const props = { extensionName: "TestExtension" };
    const { container } = render(<Extension {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("renders with extensionName and config", () => {
    const props = {
      extensionName: "TestExtension",
      config: { apiUrl: "https://example.com" },
    };
    const { container } = render(<Extension {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("renders with extensionName and stageDefinition", () => {
    const props = {
      extensionName: "TestExtension",
      stageDefinition: { build: ["compile"] },
    };
    const { container } = render(<Extension {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("renders with extensionName, config, and stageDefinition", () => {
    const props = {
      extensionName: "TestExtension",
      config: { apiUrl: "https://example.com" },
      stageDefinition: { build: ["compile"] },
    };
    const { container } = render(<Extension {...props} />);
    expect(container).toMatchSnapshot();
  });
});
