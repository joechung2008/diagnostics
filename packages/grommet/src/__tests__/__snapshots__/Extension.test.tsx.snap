// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Extension > renders with extensionName and config 1`] = `
<div>
  <div
    class="extension-root grow"
  >
    <h2
      class="StyledHeading-sc-1rdh4aw-0 fEbPmv"
    >
      TestExtension
    </h2>
    <div>
      <h3
        class="StyledHeading-sc-1rdh4aw-0 hrdewH"
      >
        Configuration
      </h3>
      <table
        aria-label="Configuration"
        class="StyledTable-sc-1m3u5g-6 wtbPz"
      >
        <thead
          class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 cYkJuW"
              scope="col"
            >
              Key
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 cYkJuW"
              scope="col"
            >
              Value
            </th>
          </tr>
        </thead>
        <tbody
          class="StyledTable__StyledTableBody-sc-1m3u5g-3"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 gBRlEM"
            >
              apiUrl
            </td>
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 gBRlEM"
            >
              https://example.com
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Extension > renders with extensionName and stageDefinition 1`] = `
<div>
  <div
    class="extension-root grow"
  >
    <h2
      class="StyledHeading-sc-1rdh4aw-0 fEbPmv"
    >
      TestExtension
    </h2>
    <div>
      <h3
        class="StyledHeading-sc-1rdh4aw-0 hrdewH"
      >
        Stage Definitions
      </h3>
      <table
        aria-label="Stage Definitions"
        class="StyledTable-sc-1m3u5g-6 wtbPz"
      >
        <thead
          class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 cYkJuW"
              scope="col"
            >
              Key
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 cYkJuW"
              scope="col"
            >
              Value
            </th>
          </tr>
        </thead>
        <tbody
          class="StyledTable__StyledTableBody-sc-1m3u5g-3"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 gBRlEM"
            >
              build
            </td>
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 gBRlEM"
            >
              compile
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Extension > renders with extensionName, config, and stageDefinition 1`] = `
<div>
  <div
    class="extension-root grow"
  >
    <h2
      class="StyledHeading-sc-1rdh4aw-0 fEbPmv"
    >
      TestExtension
    </h2>
    <div>
      <h3
        class="StyledHeading-sc-1rdh4aw-0 hrdewH"
      >
        Configuration
      </h3>
      <table
        aria-label="Configuration"
        class="StyledTable-sc-1m3u5g-6 wtbPz"
      >
        <thead
          class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 cYkJuW"
              scope="col"
            >
              Key
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 cYkJuW"
              scope="col"
            >
              Value
            </th>
          </tr>
        </thead>
        <tbody
          class="StyledTable__StyledTableBody-sc-1m3u5g-3"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 gBRlEM"
            >
              apiUrl
            </td>
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 gBRlEM"
            >
              https://example.com
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div>
      <h3
        class="StyledHeading-sc-1rdh4aw-0 hrdewH"
      >
        Stage Definitions
      </h3>
      <table
        aria-label="Stage Definitions"
        class="StyledTable-sc-1m3u5g-6 wtbPz"
      >
        <thead
          class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 cYkJuW"
              scope="col"
            >
              Key
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 cYkJuW"
              scope="col"
            >
              Value
            </th>
          </tr>
        </thead>
        <tbody
          class="StyledTable__StyledTableBody-sc-1m3u5g-3"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 gBRlEM"
            >
              build
            </td>
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 gBRlEM"
            >
              compile
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Extension > renders with only extensionName 1`] = `
<div>
  <div
    class="extension-root grow"
  >
    <h2
      class="StyledHeading-sc-1rdh4aw-0 fEbPmv"
    >
      TestExtension
    </h2>
  </div>
</div>
`;
